### PR TITLE
Fixed test, mapper tests added, code fixes

### PR DESCRIPTION
**Pull request by kvq**

### CODE REVIEW

File: src/test/java/me/kvq/HospitalTask/controller/DoctorsControllerTest.java [hunk 1]
```diff
@@ -1,162 +1,133 @@
...
                 + "\"position\":\"Position2\"}";
 
+        when(doctorService.update(eq(1L),any(DoctorDto.class))).thenAnswer(invocation -> {
+                    long dtoid = invocation.getArgument(0,Long.class);
+                    DoctorDto dto = invocation.getArgument(1,DoctorDto.class);
+                    dto.setId(dtoid);
+                    return dto;
+        });
+
         mockMvc.perform(patch("/doctor/edit/" + id)
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/controller/DoctorsControllerTest.java:79]: when-> then should be simple, without any invocation 
when(class.method(param)).thenReturn(returnValue);

File: src/test/java/me/kvq/HospitalTask/controller/DoctorsControllerTest.java [hunk 2]
```diff
@@ -1,162 +1,133 @@
...
-
-       assertEquals("Position2", serviceDoctorGet(testDoctorDto.getId()).getPosition());
+                        .andExpect(status().isOk())
+                .andExpect(content().json("{\"firstName\":\"First_NewName\","
+                        + "\"lastName\":\"Second_NewName\","
+                        + "\"patronymic\":\"Patronymic\","
+                        + "\"birthDate\":[2001,2,3],"
+                        + "\"phoneNumber\":\"381234567891\","
+                        + "\"position\":\"Position2\"}"))
+                       .andExpect(jsonPath("$.id").value(1))
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/controller/DoctorsControllerTest.java]: you should not compare full json with string, you have tests for each field, its enough

File: src/test/java/me/kvq/HospitalTask/controller/PatientControllerTest.java [hunk 3]
```diff
@@ -1,170 +1,127 @@
...
                 + "\"lastName\":\"Second_name\","
-                + "\"fathersName\":\"Fathers_Name\","
-                + "\"birthDate\":[2000,1,1],"
+                + "\"patronymic\":\"Patronymic\","
+                + "\"birthDate\":[2001,2,3],"
                 + "\"phoneNumber\":\"381234567890\","
-                + "\"doctor\":" + doctorId + "}";
+                + "\"doctor\":" + testDoctorId + "}";
+
+        when(patientService.add(any(PatientDto.class))).thenAnswer(invocation -> {
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/controller/PatientControllerTest.java]: when.thenReturn

File: src/test/java/me/kvq/HospitalTask/controller/PatientControllerTest.java [hunk 4]
```diff
@@ -1,170 +1,127 @@
...
-                "380123856789","DoctorB_Position");
-        PatientDto testPatientDto = new PatientDto(1,"PatientA_Name","PatientA_LastName", "PatientA_Patronymic",
-                LocalDate.of(1991,5,4),
-                "380123455789", testDoctorDto.getId());
-        servicePatientAdd(testPatientDto);
-
-        long id = testPatientDto.getId();
-        mockMvc.perform(delete("/patient/delete/" + id))
+        long testPatientId = 1;
+        when(patientService.delete(1L)).thenReturn(true);
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/controller/PatientControllerTest.java]: why don't you use testPatientId here?

File: src/test/java/me/kvq/HospitalTask/mapper/PatientMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+@SpringBootTest
+public class PatientMapperTest {
+    @Mock
+    DoctorDao doctorDao;
+    PatientMapper mapper;
+
+    @BeforeEach
+    void setPatientMapper(){
+        mapper = new PatientMapper(doctorDao);
+    }
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/mapper/PatientMapperTest.java]: do you really need it before each test?

File: src/test/java/me/kvq/HospitalTask/mapper/PatientMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        assertEquals(returnPatient.getDoctor(),testDoctor);
+    }
+
+    @Test
+    @DisplayName("(entityListToDtoList) passes Patient list, checks returned list size & compare Dtos")
+    void testMappingEntityListToDtoList(){
+        Doctor testDoctorA = new Doctor(3,
+                "DoctorA_Name","DoctorA_LastName", "DoctorA_Patronymic",
+                LocalDate.of(1991,5,4),
+                "380123455789", "DoctorA_Position");
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/mapper/PatientMapperTest.java:79]: doctor in patient test?

File: src/test/java/me/kvq/HospitalTask/service/DoctorServiceTest.java [hunk 7]
```diff
@@ -1,161 +1,147 @@
...
-
-    public boolean existsById(long id){
-        return storage.containsKey(id);
-    }
-
-    public Doctor getDoctorById(long id){
-        return storage.get(id);
+    void resetMocks(){
+        service = new DoctorService(doctorDao,doctorMapper);
     }
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/service/DoctorServiceTest.java]: why do you use this here?

File: src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java [hunk 8]
```diff
@@ -1,193 +1,165 @@
...
-    }
-    
-    public boolean existsById(long id){
-        return storage.containsKey(id);
-    }
-    
-    public Patient getPatientById(long id){
-        return storage.get(id);
+        PatientMapper patientMapper = new PatientMapper(doctorDao);
+        service = new PatientService(patientDao,patientMapper);
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java]: do you need to do this manually?

File: src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java [hunk 9]
```diff
@@ -1,193 +1,165 @@
...
     }
 
     @Test
-    @DisplayName("Creating new valid PatientDto, checking return Dto not null & entity exists in mocked dao")
+    @DisplayName("(add) Pass Valid Dto, then compare return Dto to passed, and checks id not same")
     void serviceAddNewValidPatientTest(){
         Doctor testDoctor = new Doctor(3,"DoctorA_Name","DoctorA_LastName", "DoctorA_Patronymic",
                 LocalDate.of(1991,5,4),
                 "380123455789","DoctorA_Position");
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java:38]: can we test without doctor?

File: src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java [hunk 10]
```diff
@@ -1,193 +1,165 @@
...
-        setTestDoctor(testDoctor);
-        Patient testPatient = new Patient(1,"PatientA_Name","PatientA_LastName", "PatientA_Patronymic",
+        PatientDto testPatient = new PatientDto(1,"PatientA_Name","PatientA_LastName", "PatientA_Patronymic",
                 LocalDate.of(1991,5,4),
-                "380123455789",testDoctor);
-        addPatients(testPatient);
+                "380123455789",testDoctor.getId());
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java:72]: this test really works?

File: src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java [hunk 11]
```diff
@@ -1,193 +1,165 @@
...
+        assertEquals(testPatient.getPatronymic(),returnedPatient.getPatronymic());
+        assertEquals(testPatient.getBirthDate(),returnedPatient.getBirthDate());
+        assertEquals(testPatient.getPhoneNumber(),returnedPatient.getPhoneNumber());
+        assertEquals(testPatient.getDoctor(),returnedPatient.getDoctor());
     }
 
     @Test
-    @DisplayName("Deleting existing patient, checking return is true & dao size decreased")
+    @DisplayName("(delete) Pass Id, expected to return true")
     void serviceDeleteExistingPatientByIdTest(){
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java]: do you need service word here?

File: src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java [hunk 12]
```diff
@@ -1,193 +1,165 @@
...
+                "PatientA_Name","PatientA_LastName", "PatientA_Patronymic",
                 LocalDate.of(1991,5,4),
-                "380123455789",testDoctor);
-        addPatients(testPatient);
-        assertEquals(1, patientDao.findAll().size(), "Test patient wasn't added to mocked dao");
+                "380123455789",null);
+        doNothing().when(patientDao).deleteById(1L);
+        when(patientDao.existsById(1L)).thenReturn(true);
 
         assertTrue(service.delete(testPatient.getId()), "Patient wasn't deleted");
```
Comment (shelsoftensy) [inline src/test/java/me/kvq/HospitalTask/service/PatientServiceTest.java]: what do you test here?

**Review by shelsoftensy — CHANGES_REQUESTED** [review]

File: src/test/java/me/kvq/HospitalTask/mapper/PatientMapperTest.java (hunk 6 above)
Comment (kvq) [reply src/test/java/me/kvq/HospitalTask/mapper/PatientMapperTest.java:79]: > doctor in patient test?

Doctor used in Patient. Even though it is optional, isn't it necessary to check if it properly maps doctor to doctor id?

**Review by shelsoftensy — APPROVED** [review]

**Pull request merged**